app/src/main/java/com/dessalines/thumbkey/db/AppDb.kt

```diff
@@ -62,15 +65,6 @@ const val DEFAULT_KEY_MODIFICATIONS = ""
 @Entity
 data class AppSettings(
     @PrimaryKey(autoGenerate = true) val id: Int,
-    @ColumnInfo(
-        name = "key_size",
-        defaultValue = DEFAULT_KEY_SIZE.toString(),
-    )
-    val keySize: Int,
-    @ColumnInfo(
-        name = "key_width",
-    )
-    val keyWidth: Int?,
```

Comment:
Your new fields need to be in the same order as before. You still need to have the defunct columns defined in the data class. It should exactly match the DB.

Reply:
The migrations should be run first, so you should be able to use the new names. Changing column order or removing columns is usually what breaks it.

Reply:
Okay purely ignoring them wasn't a good idea because then *new* settings tables have a different schema than old ones, but I think I worked out how to actually ignore and deprecate them while still keeping the schema intact.